packages/server/src/threads/automation.ts

```diff
@@ -480,7 +470,9 @@ class Orchestrator {
               success: true,
             }
           : {
-              success: true,
+              success: reachedMaxIterations
+                ? AutomationStepStatus.MAX_ITERATIONS
+                : true,
               items: this.loopStepOutputs,
```

Comment:
@imnaveenk What I think we should here do is keep `success` to be a boolean. And if `reachedMaxIterations` is true then conditionally add `status: AutomationStepStatus.MAX_ITERATIONS `

Reply:
@PClmnt resolved. I have added no iteration in condition as well this is debatable as for max_iterations=0 this case would arise. I feel its better to return no_iterations in that case.